src/core/ui/forms/MarkdownInput/MarkdownInput.tsx

```diff
@@ -0,0 +1,170 @@
+  '& p:first-child': { marginTop: 0 },
+  '& p:last-child': { marginBottom: 0 },
+  '& img': { maxWidth: '100%' },
+} as const;
+
+export const MarkdownInput = forwardRef<MarkdownInputRefApi, MarkdownInputProps>(
+  ({ value, onChange, onFocus, onBlur }, ref) => {
+    const containerRef = useRef<HTMLDivElement>(null);
+
+    const [hasFocus, setHasFocus] = useState(false);
+    const [isControlsDialogOpen, setIsControlsDialogOpen] = useState(false);
+    const isInteractingWithInput = hasFocus || isControlsDialogOpen;
+
+    const [htmlContent, setHtmlContent] = useState('');
+
+    const { markdownToHTML, HTMLToMarkdown } = usePersistentValue(UnifiedConverter());
+
+    const updateHtmlContent = async () => {
+      const content = await markdownToHTML(value);
+      setHtmlContent(String(content));
+    };
+
+    const editor = useEditor(
+      {
+        extensions: [StarterKit, ImageExtension, Link],
+        content: htmlContent,
+      },
+      [htmlContent]
+    );
+
+    useLayoutEffect(() => {
+      if (!editor || !isInteractingWithInput || editor.getText() === '') {
+        updateHtmlContent();
+      }
+    }, [value, hasFocus]);
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        focus: () => editor?.commands.focus(),
+      }),
+      [editor]
+    );
+
+    const setCharacterCount = useSetCharacterCount();
+
+    useLayoutEffect(() => {
+      setCharacterCount(editor?.getText().length ?? 0);
+    }, [editor]);
+
+    const emitChangeOnEditorUpdate = (editor: Editor) => {
+      const handleStateChange = async () => {
+        const markdown = await HTMLToMarkdown(editor.getHTML());
+
+        setCharacterCount(editor.getText().length);
+
+        onChange?.({
+          currentTarget: {
+            value: markdown,
+          },
+          target: {
+            value: markdown,
+          },
+        } as unknown as FormEvent<HTMLInputElement>);
+      };
+
+      editor.on('update', handleStateChange);
+
+      return () => {
+        editor.off('update', handleStateChange);
+      };
+    };
+
+    useEffect(() => {
+      if (editor) {
+        return emitChangeOnEditorUpdate(editor);
+      }
+    }, [editor]);
+
+    const [prevEditorHeight, setPrevEditorHeight] = useState(0);
+
+    const keepScrollPositionOnEditorReset = (editor: Editor) => {
+      const handleCreate = () => {
+        setPrevEditorHeight(0);
+      };
+
+      const handleDestroy = () => {
+        setPrevEditorHeight(editor.view.dom.clientHeight);
+      };
+
+      editor.on('create', handleCreate);
+      editor.on('destroy', handleDestroy);
+
+      return () => {
+        editor.off('create', handleCreate);
+        editor.off('destroy', handleDestroy);
+      };
+    };
+
+    useEffect(() => {
+      if (editor) {
+        return keepScrollPositionOnEditorReset(editor);
+      }
+    }, [editor]);
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const handleFocus = (event: React.FocusEvent<any>) => {
```

Comment:
I haven't been able to, but isn't there an clean way to do this without `any` ? with unknown or with HTMLElement ... 
I see the Props at the top are `InputBaseComponentProps`, maybe removing the onFocus onBlur properties... 
can end up uglier, no worries.